Improve the structure of CI

Ying1123 opened the pull request:
- Add a new file `run_suite.py` to run many `test_*.py` in a single command, so we do not need to update `.github/workflows/unit-test.yml` every time to add a new test.
- Move `deprecated` folder out of the `test` folder
- Port two more evals from openai/simple-eval: math and gpqa

Ying1123 committed:
update

Ying1123 committed:
lint

Ying1123 committed:
remove download mmlu

Ying1123 committed:
lint

Ying1123 committed:
move files

min-xu-et commented:
maybe add a readme.md under test dir to tell people how to run tests on their local machines?

Ying1123 committed:
test torch compile and chunked prefill

Ying1123 committed:
lint

Ying1123 commented:
Great suggestion! I removed the deprecated `test_process.md` and added a new file here to list the test instructions.
<img width="424" alt="image" src="https://github.com/user-attachments/assets/748dc66a-f7a0-4d05-b177-8c4858a7e5ef">

Ying1123 committed:
add more tests

Ying1123 committed:
update
